app/controllers/course/assessment/submissions_controller.rb

```diff
@@ -24,6 +24,14 @@ def edit
   end
 
   def update
+    if params[:submit_answer]
```

Comment:
I think what can simplify this is, instead of evaluating an entire submission when the submission becomes submitted, our answers are graded the moment they are submitted. Then there's no need for this complex decision on what to do.